### PR TITLE
Update the NodeDeleteAndReplace to handle master and worker nodes

### PR DESCRIPTION
**Pull request by susesgartner**
## Problem
The etcd_node_delete_and_replace_test and its helper functions where etcd specific, I updated them to be more generic and cover the above issue. 
 
## Solution
Updated the code to handle worker, controlplane, and etcd roles. 
 
## Testing
local testing Pass. 
Jenkins testing will be run.

### CODE REVIEW

File: tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go [hunk 1]
```diff
@@ -89,41 +91,46 @@ func (e *EtcdNodeDeleteAndReplace) TestEtcdNodeDeletionAndReplacement() {
...
+						if machine.Labels[nodeLabel] == "true" && machine.Labels[clusterLabel] == clusterResp.Name {
+							if machine.Name != previousNodeName {
+								nodeToDelete = machine
+							}
+							numNodesBeforeDeletions++
+						}
 					}
-				}
+					previousNodeName = nodeToDelete.Name
+					logrus.Info(fmt.Sprintf("Deleting a %s node: %s ...", nodeLabel, nodeToDelete.Name))
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go:108]: nit; have it read `Deleting %s node: %s ..."`

File: tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go [hunk 2]
```diff
@@ -89,41 +91,46 @@ func (e *EtcdNodeDeleteAndReplace) TestEtcdNodeDeletionAndReplacement() {
...
+					err = e.client.Steve.SteveType(machineSteveResourceType).Delete(&nodeToDelete)
+					require.NoError(e.T(), err)
 
-				logrus.Info("Deleting, " + etcdNodeToDelete.Name + " etcd node..")
-				err = e.client.Steve.SteveType(machineSteveResourceType).Delete(&etcdNodeToDelete)
-				require.NoError(e.T(), err)
+					clusterId, err := clusters.GetClusterIDByName(client, clusterResp.Name)
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go:112]: nit; let's change `clusterId` to be `clusterID` to be more consistent with other tests that we have in the framework.

File: tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go [hunk 3]
```diff
@@ -89,41 +91,46 @@ func (e *EtcdNodeDeleteAndReplace) TestEtcdNodeDeletionAndReplacement() {
...
+					require.NoError(e.T(), err)
+					require.True(e.T(), isEtcdNodeReplaced)
 
-				podResults, podErrors := pods.StatusPods(client, clusterId)
-				assert.NotEmpty(e.T(), podResults)
-				assert.Empty(e.T(), podErrors)
+					podResults, podErrors := pods.StatusPods(client, clusterId)
+					assert.NotEmpty(e.T(), podResults)
+					assert.Empty(e.T(), podErrors)
+				}
```
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go:128]: If you look in the `provisioning_node_driver.go`, we have a couple of additional checks we do. Can you add the checks for `CheckServiceAccountTokenSecret` and the PSACT if condition?

I know those checks currently aren't in, but I think it's good to have them here. That way, there's more conformance amongst our test cases for general validations we do manually.

**Review by markusewalker — COMMENTED** [review]
Left some comments, overall looks good to start. I'd like to see a freeform job here as well

File: tests/framework/extensions/nodes/node_status.go [hunk 4]
```diff
@@ -81,31 +81,31 @@ func IsRKE1EtcdNodeReplaced(client *rancher.Client, etcdNodeToDelete management.
...
 			return false, err
 		}
 
-		numOfEtcdNodesAfterDeletion = 0
+		numNodesAfterDeletion = 0
 		for _, machine := range machines.Data {
-			if machine.Labels[etcdLabel] == "true" && machine.Labels[clusterLabel] == clusterName {
-				if machine.Name == etcdNodeToDelete.Name {
+			if machine.Labels[nodeLabel] == "true" && machine.Labels[clusterLabel] == clusterName {
+				logrus.Info(machine.Name)
```
[the remark this answers is not in the document]
Comment (susesgartner) [reply tests/framework/extensions/nodes/node_status.go:95]: I'll update the log message to make it more clear.

File: tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go (hunk 3 above)
Comment (susesgartner) [reply tests/v2/validation/provisioning/rke2/etcd_node_delete_and_replace_test.go:128]: I'll get them in after they are merged. For now i'll add a comment as a reminder.

**Comment by caliskanugur** [comment]
@slickwarren's PR should've addressed this, closing this PR

**Pull request closed without merging**